CMSgov/cmcs-eregulations
# EREGCSC-1536 Keyboard navigability on resources page

Resolves [EREGCSC-1536](https://jiraent.cms.gov/browse/EREGCSC-1536)

**Description**

The Resources page needs some TLC to get it closer to/within 508 and screen reader compliance.  This PR will ensure that all parts of the page are accessible via keyboard navigation.

**This pull request changes:**

- Adds Skip to Main Content tab-navigable button to Resources page
- Adds and updates Cypress fixtures for Resources API call results:
   - adds fixture that mocks no/zero results
   - updates fixture that mocks page 1 of paginated results
   - adds fixture that mocks page 2 of paginated results
- Re-enables all existing tests in Resources test suite that were previously being skipped
- Adds additional resources test to provide coverage for a11y, pagination, loading state, and empty state
- adds `clearIndexedDB` Cypress command to reliably clear all cached network calls between tests.  This will be very useful.
- Refactors several components used in the Resources view to fix serious and critical a11y issues and to improve keyboard navigability
- Reformats code with prettier and corrects various eslint warnings

**Steps to manually verify this change:**

1. Visit [experimental deployment](https://bstpsibijc.execute-api.us-east-1.amazonaws.com/dev564) or run locally
2. Visit resources page and use keyboard to tab through and interact with the resources page
   - Note: if you're on a Mac, you will need to explicitly enable keyboard navigation if you have not already done so
   - https://dequeuniversity.com/mac/keyboard-access-mac
3. Run tests locally or view test results in Github actions
4. Use a11y dev tools plugin like https://www.deque.com/axe/devtools/ to ensure no critical or serious a11y issues are found
